node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -139,32 +137,33 @@ fn collect_externals_from_multi_config(
 pub fn server_initializer_collected_params<'a>(
     dirs_wrapper: &dyn DirsWrapper,
     args: &[String],
-) -> Result<GatheredParams<'a>, ConfiguratorError> {
+) -> Result<MultiConfig<'a>, ConfiguratorError> {
     let app = app_node();
+    let (config_file_path,
+        config_user_specified,
+        data_directory,
+        data_directory_specified,
+        real_user,
+        real_user_specified,
+        pre_orientation_args) = determine_user_specific_data(dirs_wrapper, &app, args)?;
+    let mut full_multi_config_vec: Vec<Box<dyn VirtualCommandLine>> = vec![
+            Box::new(EnvironmentVcl::new(&app)),
+            pre_orientation_args
+        ];
 
-    let (config_file_path, user_specified, data_directory, real_user) =
-        determine_fundamentals(dirs_wrapper, &app, args)?;
-
-    let config_file_vcl = match ConfigFileVcl::new(&config_file_path, user_specified) {
-        Ok(cfv) => Box::new(cfv),
-        Err(e) => return Err(ConfiguratorError::required("config-file", &e.to_string())),
+    let mut fill_specified_or_unspecified_box = |key: String, value: String, specified: bool | {
+        match specified {
+            true => full_multi_config_vec.push(Box::new(CommandLineVcl::new(vec!["".to_string(), key, value,]))),
+            false => full_multi_config_vec.push(Box::new(ComputedVcl::new(vec!["".to_string(), key, value,])))
+        };
     };
-    let full_multi_config = make_new_multi_config(
-        &app,
-        vec![
-            Box::new(CommandLineVcl::new(args.to_vec())),
-            Box::new(EnvironmentVcl::new(&app)),
-            config_file_vcl,
-        ],
-    )?;
-    let config_file_path =
-        value_m!(full_multi_config, "config-file", PathBuf).expect("defaulted param");
-    Ok(GatheredParams::new(
-        full_multi_config,
-        config_file_path,
-        real_user,
-        data_directory,
-    ))
+    fill_specified_or_unspecified_box("--config-file".to_string(), config_file_path.to_string_lossy().to_string(), config_user_specified);
+    fill_specified_or_unspecified_box("--data-directory".to_string(), data_directory.to_string_lossy().to_string(), data_directory_specified);
+    fill_specified_or_unspecified_box("--real-user".to_string(), real_user.to_string(), real_user_specified);
```

Comment:
Again, a function call usually look better if without other function calls in the place of its arguments.  You can enhance it here if you supply `&str` instead of the first argument. You will simply call the `to_string()` method inside the function.